src/picongpu/include/simulationControl/GuardHandlerCallPlugins.hpp

```diff
@@ -0,0 +1,64 @@
+/**
+ * Copyright 2016 Heiko Burau
+ *
+ * This file is part of PIConGPU.
+ *
+ * PIConGPU is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * PIConGPU is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with PIConGPU.
+ * If not, see <http://www.gnu.org/licenses/>.
+ */
+
+#pragma once
+
+#include "particles/policies/ExchangeParticles.hpp"
+#include "pluginSystem/IPlugin.hpp"
+#include <list>
+
+namespace picongpu
+{
+
+/**
+ * Guard handler policy calling all registered plugins when particles
+ * leave the global simulation volume. This class serves as policy for
+ * the `ParticleDescription` template class.
+ *
+ * For each plugin the method `IPlugin::onParticleLeave()` is called.
+ * After that the guard particles are deleted.
+ */
+struct GuardHandlerCallPlugins
+{
+    typedef PMacc::particles::policies::ExchangeParticles HandleExchanged;
+    typedef GuardHandlerCallPlugins HandleNotExchanged;
+
+    template< class T_Particles >
+    void
+    handleOutgoing(T_Particles& particles, int32_t direction) const
+    {
+        typedef std::list<PMacc::IPlugin*> Plugins;
+        Plugins plugins = Environment<>::get().PluginConnector().getAllPlugins();
+
+        for(Plugins::iterator iter = plugins.begin(); iter != plugins.end(); iter++)
+        {
+            (*iter)->onParticleLeave(T_Particles::FrameType::getName(), direction);
+        }
+
+        particles.deleteGuardParticles(direction);
+    }
+
+    template< class T_Particles >
+    void
+    handleIncoming(T_Particles& par, int32_t direction) const
```

Comment:
pls comment out the unused `par` and `direction` (or remove the args) :) otherwise some compilers throw warnings